### PR TITLE
Feat/lightgbm parallelization

### PR DESCRIPTION
**Pull request by AleksMat**
Changes:
- Added utility function `execute_with_multiprocessing_lock` into `eolearn.core` and unit tests for it.
- Fixed cloud masking tasks to load LightGBM models lazily.
- Fixed cloud masking tasks to predict with LightGBM models using `execute_with_multiprocessing_lock`.

Note: this is related to PR https://github.com/sentinel-hub/sentinel2-cloud-detector/pull/18

### CODE REVIEW

File: core/eolearn/tests/test_eoexecutor.py [hunk 1]
```diff
@@ -167,5 +170,60 @@ def test_exceptions(self):
...
+
+            handler.close()
+            logger.removeHandler(handler)
+
+            with open(fp.name, 'r') as log_file:
+                lines = log_file.read().strip('\n ').split('\n')
+
+            self.assertEqual(len(lines), 2 * self.WORKERS)
+            self.assertEqual(len(set(lines[: self.WORKERS])), self.WORKERS, msg='All processes should start')
+            self.assertEqual(len(set(lines[self.WORKERS: ])), self.WORKERS, msg='All processes should finish')
```
Comment (mlubej) [inline core/eolearn/tests/test_eoexecutor.py]: extra whitespace

**Comment by mlubej** [comment]
Can you add `hypothesis` to core requirements? It's used in `test_eoworkflow.py`

**Comment by AleksMat** [comment]
> Can you add `hypothesis` to core requirements? It's used in `test_eoworkflow.py`

`hypotesis` is already in [development requirements](https://github.com/sentinel-hub/eo-learn/blob/master/requirements-dev.txt). The main requirements are only meant for running the package, while development requirements are additional things for running unit tests, linting and overall package development.

**Comment by mlubej** [comment]
> > Can you add `hypothesis` to core requirements? It's used in `test_eoworkflow.py`
> 
> `hypotesis` is already in [development requirements](https://github.com/sentinel-hub/eo-learn/blob/master/requirements-dev.txt). The main requirements are only meant for running the package, while development requirements are additional things for running unit tests, linting and overall package development.

Ah sorry, missed it..

File: core/eolearn/tests/test_eoexecutor.py (hunk 1 above)
Comment (AleksMat) [reply core/eolearn/tests/test_eoexecutor.py]: fixed

**Pull request merged**